src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -911,8 +928,9 @@ private void StartStream()
                 _currentHeadersStream.OnEndStreamReceived();
             }
 
-            _activeStreamCount++;
             _streams[_incomingFrame.StreamId] = _currentHeadersStream;
+            _currentHeadersStream.IncrementActiveStreamCount();
```

Comment:
I don't like how I had to call into the Http2Stream, which calls back into this class, to increment/decrement. Each stream needs to guarantee it's only incremented/decremented once. Increment should be called once, but decrement can be called twice.

Reply:
Incrementing isn't doing anything special, only decrementing needs to be routed through the stream.

We don't have any scenarios where a stream can try to decrement if it was never incremented, right?

Reply:
I think you are right.